cmd/cluster/support/delete.go

```diff
@@ -120,15 +116,22 @@ func (o *deleteOptions) run() error {
 	}
 
 	var limitedSupportReasonIds []string
+	limitedSupportReasons, err := getLimitedSupportReasons(o.clusterID)
+
+	if len(limitedSupportReasons) == 0 {
+		fmt.Fprintf(os.Stderr, "Cluster does not has any limited support reason ID \n")
+		os.Exit(1)
+	}
+
 	if o.removeAll {
-		limitedSupportReasons, err := getLimitedSupportReasons(o.clusterID)
 		for _, limitedSupportReason := range limitedSupportReasons {
 			limitedSupportReasonIds = append(limitedSupportReasonIds, limitedSupportReason.ID())
 		}
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "Failed to get limited support reasons: %v\n", err)
-			return err
-		}
+	} else if len(limitedSupportReasons) > 1 && o.limitedSupportReasonID == "" {
+		fmt.Fprintf(os.Stderr, "This cluster has multiple limited support reason IDs.\nPlease specify the exact reason ID or the `all` flag \n")
+		os.Exit(1)
+	} else if len(limitedSupportReasons) == 1 && o.limitedSupportReasonID == "" {
+		limitedSupportReasonIds = append(limitedSupportReasonIds, limitedSupportReasons[0].ID())
 	} else {
 		limitedSupportReasonIds = append(limitedSupportReasonIds, o.limitedSupportReasonID)
```

Comment:
**question:** this should be the same, right? 

I think we could do the following to make the logic easier to understand:
- `o.removeAll` and `(len(limitedSupportReasonIds) > 1 && o.limitedSupportReasonID == "")` is the same state, merge them
- merge the above two else's into one as well (they should be the same)
- move the deletion logic into a new function `deleteLimitedSupportReason` 

E.g. (pseudocode)
```
if len(limitedSupportReasonIds) == 0 {
    return errors.New("cluster is not in limited support")
}

if o.removeAll || (len(limitedSupportReasonIds) > 1 && o.limitedSupportReasonID == "") {
    if !o.removeAll && o.limitedSupportReasonID == "" {
        return errors.New("cluster has multiplate limited support reason IDs, please specify the exact reason ID or use the `all` flag.")
    }

    for _, id := range limitedSupportReasonIds {
        deleteLimitedSupportReason(connection, cluster, id)
    }
} else {
    limitedSupportReasonIds = append(limitedSupportReasonIds, o.limitedSupportReasonID)
    for _, id := range limitedSupportReasonIds {
        deleteLimitedSupportReason(connection, cluster, id)
    }
}
```

Reply:
The if..else is now consolidated and deleteLimitedSupportReason() is also implemented. 
However, I am still unable to delete the "only" LS reason ID in the absence of the -i flag.

Reply:
At a quick glance I'm assuming you're running into an error in my pseudocode :D 
```
} else {
    limitedSupportReasonIds = append(limitedSupportReasonIds, o.limitedSupportReasonID)
    for _, id := range limitedSupportReasonIds {
        deleteLimitedSupportReason(connection, cluster, id)
    }
```

The limitedSupportReasonIds would be empty in that case, it needs to actually be filled with the single reason (we can add a check for safety that it indeed is only one). You can get that ID from `limitedSupportReasons, err := getLimitedSupportReasons(o.clusterID)` called earlier.

Reply:
this works now.